javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll

```diff
@@ -189,4 +189,25 @@ module UnsafeShellCommandConstruction {
       )
     }
   }
+
+  /**
+   * A check of the form `type x === "X"`, where x is  "number", or "boolean",
+   * which sanitized `x` in its "then" branch.
+   */
+  class TypeOfSanitizer extends TaintTracking::SanitizerGuardNode, DataFlow::ValueNode {
```

Comment:
(It could be useful to have this as a public class in `TaintTracking.qll`, perhaps configurable. `UnsafeJQueryPluginCustomizations` could also use it in its treatment of `typeof` guards.)

Reply:
Turns out we have a lot of places with a `typeof` guard.  
I made a `TaintTracking::isTypeofGuard` predicate, and used that throughout our project.

Reply:
great initiative. Lets wait proper performance evaluation of that change then.

Reply:
Looks fine https://git.semmle.com/erik/dist-compare-reports/tree/eriks-profile-machine_1608843110557